### PR TITLE
[Fix] fix the contract test isolation

### PR DESCRIPTION
**Pull request by Lisooo790926**
Please follow the guidelines in [Contribution.md](https://github.com/social-tw/social-tw-website/blob/main/CONTRIBUTING.md) first, then start to create your pull request here.

## Summary
Fix contract test random error by adding isolation between test

## Details
1. add evm_revert after test 
2. add each userState stop after finish each test

## Impacted Areas
Contract

## Tests
Contract

## Checklist

-   [x] All new and existing tests pass
-   [x] I have updated the documentation (if applicable)
-   [x] My changes do not introduce new security vulnerabilities
-   [x] Run `yarn lint:fix`

### CODE REVIEW

**Review by ryanycw — COMMENTED** [review]
LGTM!! Do we want to be consistent between the style of `packages/contracts/test/SignupAndPost.test.ts` and `packages/contracts/test/Comment.test.ts`

**Comment by FoodChain1028** [comment]
Hi, @Lisooo790926 could we merge this one into main now?

File: packages/contracts/test/SignupAndPost.test.ts [hunk 1]
```diff
@@ -111,13 +124,13 @@ describe('Unirep App', function () {
             // generate a fake proof
             const concoctProof = [...proof]
             const len = concoctProof[0].toString().length
-            concoctProof[0] = BigInt(
-                proof[0].toString().slice(0, len - 1) + BigInt(2)
-            )
+            concoctProof[0] = BigInt(2)
```
Comment (ryanycw) [inline packages/contracts/test/SignupAndPost.test.ts:127]: What is the change here? Besides this, everything looks good

Comment (FoodChain1028) [reply packages/contracts/test/SignupAndPost.test.ts:127]: The original concocted proof might create the same proof as the real one (but not 100% sure, the error was: `This Test should revert yet didn't`). I deem this change might lower the opportunity of collision.

Comment (ryanycw) [reply packages/contracts/test/SignupAndPost.test.ts:127]: Make sense. It should lower the opportunity. But it's a bit weird, the original impl should be fine.

**Review by ryanycw — APPROVED** [review]

**Pull request merged**